pexpect/pty_spawn.py

```diff
@@ -430,11 +452,7 @@ def read_nonblocking(self, size=1, timeout=-1):
         # For this case, I test isalive() before doing any reading.
         # If isalive() is false, then I pretend that this is the same as EOF.
```

Comment:
The comments suggest that without this check, there's a risk of blocking on Solaris. So I'd rather not bypass it unless we're confident that this is no longer an issue when you run in Solaris.